models/apeswap/polygon/apeswap_polygon_trades.sql

```diff
@@ -0,0 +1,103 @@
+{{ config(
+    alias = 'trades'
+    ,partition_by = ['block_date']
+    ,materialized = 'incremental'
+    ,file_format = 'delta'
+    ,incremental_strategy = 'merge'
+    ,unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index', 'trace_address']
+    ,post_hook='{{ expose_spells(\'["avalanche_c"]\',
+                                      "project",
+                                      "apeswap",
+                                    \'["codingsh", "zhongyiio"]\') }}'
+    )
+}}
+
+{% set project_start_date = '2021-03-15' %}
+
+WITH apeswap_dex AS (
+    SELECT  t.evt_block_time                                             AS block_time,
+            `to`                                                         AS taker,
+            sender                                                       AS maker,
+            CASE WHEN amount0Out = 0 THEN amount1Out ELSE amount0Out END AS token_bought_amount_raw,
+            CASE WHEN amount0In = 0 THEN amount1In ELSE amount0In END    AS token_sold_amount_raw,
+            cast(NULL as double)                                         AS amount_usd,
+            CASE WHEN amount0Out = 0 THEN token1 ELSE token0 END         AS token_bought_address,
+            CASE WHEN amount0In = 0 THEN token1 ELSE token0 END          AS token_sold_address,
+            t.contract_address                                           AS project_contract_address,
+            t.evt_tx_hash                                                AS tx_hash,
+            ''                                                           AS trace_address,
+            t.evt_index
+    FROM {{ source('apeswap_v2_polygon', 'Pair_evt_Swap') }} t
+    INNER JOIN {{ source('apeswap_v2_polygon', 'SushiV2Factory_evt_PairCreated') }} p
```

Comment:
Please verify other sources on dune V2 as well, we can wait for the decoded tables first.